qemu/tests/vioinput_hotplug.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import time
+
+from virttest import error_context
+from provider import input_tests
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test hotplug/unplug of virtio input device
+    1) Boot up w/ one virtio input device
+    2) Unplug one virtio input device
+    3) Hotplug one virtio input device
+    4) Run basic keyboard/mouse test
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def hotplug_input_dev(vm, dev):
+        error_context.context("Hotplug %s" % dev, logging.info)
+        out, ver_out = vm.devices.simple_hotplug(dev, vm.monitor)
+        time.sleep(5)
```

Comment:
Would you please clarify about why we need this at here? thanks.

Reply:
For virtio-mouse/tablet device, after new device added, the default working device will change from ps/2 mice to new added mice, this progress need some time, here add 5 secs to wait this progress finish, but I'll move this line outside func hotplug_input_dev, will add it before sub_test run.

Reply:
Is there any event reported by the os that we can monitor to verify the state than sleep? if no, please add a comment to explain this line, thanks.

Reply:
We can through {"execute":"query-mice"}} to check if new added mice device starting work, but it's also a step in sub_test=mouse_test.

after hotplug completed, if {"execute":"query-mice"}} immediately， will receive follows message, new added device "QEMU Virtio Mouse", "current": false, means not work now.
{"return": [{"index": 4, "name": "QEMU Virtio Mouse", "current": false, "absolute": false}, {"index": 2, "name": "QEMU PS/2 Mouse", "current": true, "absolute": false}]}

wait 2-3 sec, execute {"execute":"query-mice"}} again, will receive follows message, "QEMU Virtio Mouse", "current": true, means it start work.

{"return": [{"index": 2, "name": "QEMU PS/2 Mouse", "current": false, "absolute": false}, {"index": 4, "name": "QEMU Virtio Mouse", "current": true, "absolute": false}]}

But in mouse_test steps, the first step is to check if new added mouse device is working, through {"execute":"query-mice"}} as well, so I think add sleep time here is ok, I will add a comment to explain this line.

Reply:
ok, thanks for your explanation.